chemprop/nn/predictors.py

```diff
@@ -178,7 +181,7 @@ def forward(self, Z: Tensor) -> Tensor:
         mean = self.output_transform(mean)
         var = self.output_transform.transform_variance(var)
 
-        return torch.cat((mean, var), 1)
+        return torch.stack((mean, var), dim=2)
```

Comment:
Is the goal to have the output of
```
feedforward = MveFFN()
feedforward(torch.rand(n_batches, n_hidden_dim))
```
be a tensor of shape (n_batches, n_tasks, n_targets = 2)?
If so we don't need to add `nn.Unflatten()` to `self.ffn` because the line `Y = self.ffn(Z)` is internal to `MveFFN`'s `forward` method. We can just return `torch.stack((mean, var), dim=2)` like you have done here. 

Example:
```
import torch

n_batch = 4
n_tasks = 1
n_targets = 2

Y = torch.arange(n_batch * n_tasks * n_targets).reshape(n_batch, n_tasks * n_targets)
Y:
tensor([[0, 1],
        [2, 3],
        [4, 5],
        [6, 7]])

mean, var = torch.chunk(Y, n_targets, 1)
mean:
tensor([[0], [2], [4], [8]])
var:
tensor([[1], [3], [5], [7]])

# do stuff with mean and var

output = torch.stack((mean, var), 2)
output[..., 0]
tensor([[0], [2], [4], [8]])
```